keras/backend/tensorflow_backend.py

```diff
@@ -570,6 +570,26 @@ def categorical_crossentropy(output, target, from_logits=False):
         return tf.nn.softmax_cross_entropy_with_logits(output, target)
 
 
+def categorical_crossentropy_one_hot(output, target, from_logits=False):
+    '''Note: tf.nn.sparse_softmax_cross_entropy_with_logits
+    expects logits, Keras expects prbabilities.
+    '''
+    if not from_logits:
+        output = tf.clip_by_value(output, tf.cast(_EPSILON, dtype=_FLOATX),
+                                  tf.cast(1.-_EPSILON, dtype=_FLOATX))
+        output = tf.log(output / (1 - output))
```

Comment:
@fchollet I don't think I agree (but I blame confusing Tensorflow docs). All the TF call does is perform a softmax, and then a normal cross-entropy calculation. So to get the right answer you need to just take a `log` of output, not do this `output / (1 - output)` thing. 

If you run this sample you will see that 

```
import tensorflow as tf
import numpy as np

sess = tf.InteractiveSession()  
prob_dist = np.asarray([[1/2, 1/4, 1/4]])
target = np.asarray([[1/3, 1/3, 1/3]])
log_prob = tf.constant(np.log(prob_dist))
log_odds = tf.constant(np.log(prob_dist / (1 - prob_dist)))
                       
manual_answer = -np.sum(target * np.log(prob_dist))
straight_log = tf.nn.softmax_cross_entropy_with_logits(log_prob, tf.constant(target)).eval()
fancy_logit  = tf.nn.softmax_cross_entropy_with_logits(log_odds, tf.constant(target)).eval()

print("Manual:    {}".format(manual_answer))
print("Logarithm: {}".format(straight_log[0]))
print("Logit:     {}".format(fancy_logit[0]))

sess.close()
```

outputs:

```
Manual:    1.1552453009332422
Logarithm: 1.1552453009332422
Logit:     1.2432338162113972
```

i.e. to get the same answer as a manual cross-entropy calculation you need to take a log, *not* perform the log-odds operation.